iroh-net/src/key.rs

```diff
@@ -50,6 +50,18 @@ impl PublicKey {
         self.public.as_bytes()
     }
 
+    /// Construct a `PublicKey` from a slice of bytes.
+    ///
+    /// # Warning
+    ///
+    /// This will return a [`SignatureError`] if the bytes passed into this method do not represent
+    /// a valid `ed25519_dalek` curve point. Will never fail for bytes return from [`Self::as_bytes`].
+    /// See [`VerifyingKey::from_bytes`] for details.
+    pub fn from_bytes(bytes: &[u8; 32]) -> Result<Self, SignatureError> {
```

Comment:
Is this really needed when we have `TryFrom<&[u8>`?

Reply:
For me having a from_bytes method seems convenient. Eg all the `ed22591_dalek` types have it as well. I can also remove it though..